src/sentry_core.c

```diff
@@ -61,7 +61,18 @@ sentry_init(sentry_options_t *options)
     sentry_shutdown();
     sentry__mutex_lock(&g_options_mutex);
     g_options = options;
+
     sentry__path_create_dir_all(options->database_path);
+    sentry_path_t *database_path = options->database_path;
+    options->database_path = sentry__path_absolute(database_path);
+    if (options->database_path) {
+        sentry__path_free(database_path);
+    } else {
+        options->database_path = database_path;
```

Comment:
_nit_: I'd add an error log here, for now, since this case indicates a major error. In the future, we should leave the database path `NULL`, and handle that explicitly in the transports, user consent, and run handling.